### PR TITLE
Create mail queue and use SendEmail job for all mailables

### PR DESCRIPTION
**Pull request by marcusmyers**
In order to use a queue when sending emails, this commit also adds the
`SendEmail` job that can be used with any Mailable object and it will
properly add the email to the queue.

[56](#56)

### CODE REVIEW

File: config/queue.php [hunk 1]
```diff
@@ -37,7 +37,7 @@
         'database' => [
             'driver' => 'database',
             'table' => 'jobs',
-            'queue' => 'default',
+            'queue' => ['mail','default'],
```
Comment (Zelig880) [inline config/queue.php:40]: Would you be able to use the default queue for the mail please. (the production site can just one the default queue at the time being!)

File: app/Http/Controllers/Ledge/ManagementController.php [hunk 2]
```diff
@@ -58,7 +53,7 @@ public function return($id)
                 'status' => LedgeStatus::AwaitingReturn
             ]);
 
-            Mail::to($ledge->lender->email)->send(new BookReturnStatusMail($ledge));
+            SendEmail::dispatch('App\Mail\BookReturnStatusMail', $ledge, $ledge->lender->email)->onQueue('mail');
```
Comment (Zelig880) [inline app/Http/Controllers/Ledge/ManagementController.php:56]: Due to the comment below (to change the queue to use default due to config issue) can you remove the `onQueue('mail')` to all this entry please :)

**Review by Zelig880 — CHANGES_REQUESTED** [review]
Hi thank you so much for your PR!!!!! Do not worry about the delay, what matter is that you made it! 

The PR looks great, but if you do not mind, it would be great to just use the default queue, as I have problem on production to set multiple queue at the time! :)

File: config/queue.php (hunk 1 above)
Comment (marcusmyers) [reply config/queue.php]: Amended in 82764e4.

File: app/Http/Controllers/Ledge/ManagementController.php (hunk 2 above)
Comment (marcusmyers) [reply app/Http/Controllers/Ledge/ManagementController.php]: Amended in 82764e4.

**Review by Zelig880 — APPROVED** [review]

**Pull request merged**